sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -658,6 +658,11 @@ object JdbcUtils extends Logging with SQLConfHelper {
       dialect: JdbcDialect,
       isolationLevel: Int,
       options: JDBCOptions): Unit = {
+
+    if (iterator.isEmpty) {
```

Comment:
I don't see a reason this will do any more work than is done now - it will have to begin evaluate the partition, but, it does that anyway once it reaches the loop below.